### ISSUE TITLE
LocalIndicesCleanerTests#testHandlesWatcherHistory fails

### ISSUE THREAD

**Issue by dnhatn** [elastic/elasticsearch#67198]
**Build scan**: https://gradle-enterprise.elastic.co/s/a4vne4xxjvqvc

**Repro line**:

```
 ./gradlew ':x-pack:plugin:monitoring:test' --tests "org.elasticsearch.xpack.monitoring.cleaner.local.LocalIndicesCleanerTests.testHandlesWatcherHistory" -Dtests.seed=C62A56FEEF0DB9C9 -Dtests.security.manager=true -Dtests.locale=uk -Dtests.timezone=Pacific/Pohnpei -Druntime.java=11
```

**Reproduces locally?**: yes

**Applicable branches**: master

**Failure excerpt**:
```
  2> java.lang.IllegalArgumentException: cannot create index with name [.watcher-history-14-2020.12.31], because it matches with template [.watch-history-14] that creates data streams only, use create data stream api instead
        at __randomizedtesting.SeedInfo.seed([C62A56FEEF0DB9C9:746BC093E3D1FF7F]:0)
        at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequestWithV2Template(MetadataCreateIndexService.java:504)
        at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:357)
        at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:377)
        at org.elasticsearch.cluster.metadata.MetadataCreateIndexService$1.execute(MetadataCreateIndexService.java:300)
        at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:59)
        at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:697)
        at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:319)
        at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:214)
        at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:151)
        at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)
        at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:680)
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252)
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by martijnvg** [comment]
This can be closed once #67154 has been merged.

**Comment by martijnvg** [comment]
Closing, #67154 has been merged and this failure should no longer occur.
